specification/communication/data-plane/Rooms/preview/2022-02-01-preview/communicationservicesrooms.json

```diff
@@ -0,0 +1,846 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Communication Room Service",
+    "description": "Azure Communication Room Service",
+    "version": "2022-02-01"
+  },
+  "paths": {
+    "/rooms": {
+      "post": {
+        "tags": [
+          "Rooms"
+        ],
+        "summary": "Creates a new room.",
+        "operationId": "Rooms_CreateRoom",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "header",
+            "name": "Repeatability-Request-ID",
+            "description": "If specified, the client directs that the request is repeatable; that is, that the client can make the request multiple times with the same Repeatability-Request-Id and get back an appropriate response without the server executing the request multiple times. The value of the Repeatability-Request-Id is an opaque string representing a client-generated, globally unique for all time, identifier for the request. It is recommended to use version 4 (random) UUIDs.",
+            "type": "string",
+            "format": "uuid"
+          },
+          {
+            "in": "header",
+            "name": "Repeatability-First-Sent",
+            "description": "If Repeatability-Request-ID header is specified, then Repeatability-First-Sent header must also be specified. The value should be the date and time at which the request was first created, expressed using the IMF-fixdate form of HTTP-date",
+            "type": "string",
+            "format": "date-time"
+          },
+          {
+            "in": "body",
+            "name": "createRoomRequest",
+            "description": "The create room request body.",
+            "schema": {
+              "$ref": "#/definitions/CreateRoomRequest"
+            },
+            "required": true
+          }
+        ],
+        "responses": {
```

Comment:
missing "default", see https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#handling-errors

and `x-ms-error-code` header is recommended in all error responses.

Reply:
Default error response added with header x-ms-error-code in all error response on the specification.
@weidongxu-microsoft I amlooking for API implementation refences for x-ms-error-code on Dot Net code bases. Any pointers.

Reply:
I am not familiar with .NET SDK. By implementation references, do you mean hand-written code or generated code (from autorest)?

I think SDK will also expose the `x-ms-error-code` in response, but at that time, the JSON in response likely already be de-serialized.
The advantage of `x-ms-error-code` would be for REST API user, that if they see 4xx, they can check the header and see whether it is recoverable (without inspecting the JSON in body).

Reply:
I was looking for backend service implementation references for services written in .NET.
On further search, was able to find a reference under communication services itself. Email services seems to have implemented x-ms-error-code header response for error responses.
https://skype.visualstudio.com/DefaultCollection/SCC/_git/ic3_spool_cosine-dep-spool.  I will refer to this for backed implementation. 
The Spec is updated as per your recommendation @weidongxu-microsoft